omc3/model/accelerators/lhc.py

```diff
@@ -212,32 +214,41 @@ def get_lhc_error_dir() -> Path:
         return LHC_DIR / "systematic_errors"
 
     def get_variables(self, frm: float = None, to: float = None, classes: Iterable[str] = None):
-        correctors_dir = LHC_DIR / "2012" / "correctors"
-        all_vars_by_class = _merge_jsons(
-            correctors_dir / f"correctors_b{self.beam}" / "beta_correctors.json",
-            correctors_dir / f"correctors_b{self.beam}" / "coupling_correctors.json",
-            self._get_triplet_correctors_file(),
+        corrector_beam_dir = Path(f"correctors_b{self.beam}")
+        all_vars_by_class = _load_jsons(
+            *self._get_corrector_files(corrector_beam_dir / "beta_correctors.json"),
+            *self._get_corrector_files(corrector_beam_dir / "coupling_correctors.json"),
+            *self._get_corrector_files("triplet_correctors.json"),
```

Comment:
So with the changes below, we now look for the corrector files that were put in the model directory and default to the hard-coded ones included with `omc3`, right?

Reply:
Almost. 
Up until now we had all the `corrector.json` files in the `models/lhc/2012/correctors` dir. And then adapted these files as we pleased.  I did not like this for two reasons:
a) The files were becoming really big
b) If we change the correctors to use at some point, e.g. remove knobs in 2024 from the  "MQM" category, because we cannot use them anymore from now on, calculating corrections for older models, e.g. 2018, become hard to trace back, because back in 2018 we used different knobs that are now present.

So now, I moved the `correctors` dir from 2012 directly to `models/lhc/correctors` and designated as "default".
In case we want to change the categories or add new ones that are specific for a year, we can place them in "year/correctors". 
Both files, default and year, will always be loaded, but because the `.json` file from the year is loaded later, it has priority in case it defines the same category name.

Reply:
That being said, I do like your misunderstanding... I could easily add a third layer, that first checks in the model directory, making it super flexible for online-analysis and avoiding the hacking at all. 
That was also a reason, why I implemented giving the knob names directly and also be able to remove them, because often the "hacks" are only small changes.